cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/utils/RandomUtils.java

```diff
@@ -63,6 +64,24 @@ public static int randomPositiveInt(int bound)
         return RANDOM.nextInt(bound - 1) + 1;
     }
 
+    public static int nextInt(final int startInclusive, final int endExclusive)
+    {
+        if (endExclusive < startInclusive)
+        {
+            throw new IllegalArgumentException("Start value must be smaller or equal to end value.");
+        }
+        if (startInclusive < 0)
+        {
+            throw new IllegalArgumentException("Both range values must be non-negative.");
+        }
+        if (startInclusive == endExclusive)
+        {
+            return startInclusive;
+        }
+
+        return startInclusive + RANDOM.nextInt(endExclusive - startInclusive);
```

Comment:
The bound value for random is exclusive. I think we are missing a value here. 

For example, when `endExclusive - startInclusive == 1`, the produce pseudo random values contains only a single value of 0, but it wants both 0 and 1 instead. 

The util is only used for testing, hence a minor issue. Let's get it correct still.

Reply:
If it is end exclusive and start inclusive and the diff is 1, then shouldn't it only return 1 value? E.g. (4, 5] should always return 4?

Reply:
Ah. Then the if block above should throw IllegalArgumentException too? e.g. [4, 4) is empty 

```java
        if (startInclusive == endExclusive)
        {
            return startInclusive;
        }
```

Reply:
It currently does this:

```
if (startInclusive == endExclusive)
{
    return startInclusive;
}
```

I don't particularly mind either way, current approach seems more permissive.